src/sagemaker/cli/framework_upgrade.py

```diff
@@ -0,0 +1,251 @@
+):
+    """Read framework image uri information from json file to a dictionary, update it with new
+    framework version information, then write the dictionary back to json file.
+
+    Args:
+        existing_content (dict): Existing framework/algorithm image uri information read from
+            json file.
+        short_version (str): Abbreviated framework version (e.g. 1.0, 1.5).
+        full_version (str): Complete framework version (e.g. 1.0.0, 1.5.2).
+        scope (str): Framework image type, it could be "training", "inference".
+        processors (str): Supported processors (e.g. "cpu,gpu").
+        py_versions (str): Supported Python versions (e.g. "py3,py37").
+        tag_prefix (str): Algorithm image's tag prefix.
+     """
+    py_versions = py_versions.split(",")
+    processors = processors.split(",")
+    latest_registries, latest_py_versions, latest_repository = _get_latest_values(
+        existing_content, scope
+    )
+    if not py_versions:
+        py_versions = latest_py_versions
+
+    if scope in existing_content:
+        add_dlc_framework_version(
+            existing_content,
+            short_version,
+            full_version,
+            scope,
+            processors,
+            py_versions,
+            latest_registries,
+            latest_repository,
+        )
+    else:
+        scopes = scope.split(",")
+        add_algo_version(
+            existing_content,
+            processors,
+            scopes,
+            full_version,
+            py_versions,
+            latest_registries,
+            latest_repository,
+            tag_prefix,
+        )
+
+
+def main():
+    """Parse command line arguments, call corresponding methods."""
+    parser = argparse.ArgumentParser(description="Framework upgrade tool.")
+    parser.add_argument(
+        "--framework", required=True, help="Name of the framework (e.g. tensorflow, mxnet, etc.)"
+    )
+    parser.add_argument("--short-version", help="Abbreviated framework version (e.g. 2.0)")
+    parser.add_argument("--full-version", help="Full framework version (e.g. 2.0.1)")
+    parser.add_argument("--processors", help="Suppoted processors (e.g. cpu, gpu)")
+    parser.add_argument("--py-versions", help="Supported Python versions (e.g. py3,py37)")
+    parser.add_argument("--scope", help="Scope for the Algorithm image (e.g. inference, training)")
+    parser.add_argument(
+        "--tag-prefix", help="Tag prefix of the Algorithm image (e.g. ray-0.8.5-torch)"
+    )
+    parser.add_argument("--region", help="New region to be added (e.g. us-west-2)")
+    parser.add_argument("--account", help="Registry account of new region")
+
+    args = parser.parse_args()
+    framework = args.framework
+    short_version = args.short_version
+    full_version = args.full_version
+    processors = args.processors
+    py_versions = args.py_versions
+    scope = args.scope
+    tag_prefix = args.tag_prefix
+    region = args.region
+    account = args.account
```

Comment:
(optional/nitpick) I think I'd probably just use the `args.<name>` directly rather than assigning new variables - I don't feel like this is providing much value here.

Reply:
make sense!